src/decorators/relationships.js

```diff
@@ -284,10 +284,10 @@ export class RelatedRecordsArray extends Array {
 
     if (relationships && relationships[property] && relatedRecord) {
       const referenceIndexToRemove = relationships[property].data.findIndex((model) => model.id === id && model.type === type)
-      relationships[property].data.splice(referenceIndexToRemove, 1)
+      if (referenceIndexToRemove >= 0) relationships[property].data.splice(referenceIndexToRemove, 1)
 
       const recordIndexToRemove = this.findIndex((model) => model.id === id && model.type === type)
-      if (recordIndexToRemove > 0) this.splice(recordIndexToRemove, 1)
+      if (recordIndexToRemove >= 0) this.splice(recordIndexToRemove, 1)
```

Comment:
If these are `-1` should we just do an early return?

It would be nice if this line wasn't called if no change was actually made
```
record._dirtyRelationships.add(property)
```

Not sure about. maybe @jondayton can comment on that? ```
      // hack this will only work with singularized relationships.
      setRelatedRecord(relatedRecord, null, recordType.slice(0, recordType.length - 1))
```

Reply:
I believe `setRelatedRecord` will be fine because if the record isn't in the relationships or related record array then it's essentially a no-op. Of course, that means it's also fine to early return. 👍

Reply:
Let's do an early return if we can, then.